src/main/java/io/github/thebusybiscuit/slimefun4/api/network/Network.java

```diff
@@ -241,6 +252,32 @@ private void discoverNeighbors(@Nonnull Location l) {
         discoverNeighbors(l, 0.0, 0.0, -1.0);
     }
 
+    private void updateNeighbors(@Nonnull Location l, double xDiff, double yDiff, double zDiff) {
+        for (int i = 1; i <= getRange(); i++) {
+            Location newLocation = l.clone().add(i * xDiff, i * yDiff, i * zDiff);
```

Comment:
Slightly confused by this. What is this method supposed to do?

As it is written I don't see it ever being useful to pass anything but one unitary value (with sign) and two zeros, like you do below, therefore the method only updatesNeighbors in one direction.
But then the question arises on whether this format even makes sense, it would be a lot more readable (though not as compact) if it accepted a BlockFace at that point.

Was the idea just to avoid having to "decode" what versor each BlockFace refers to?

Reply:
This method checks if any Locations need to be added to the network when an insulator is broken. 

It indeed is intended to only check the 6 directions in which a network can connect. The reason I went with this signature is because the method is similar to discoverNeighbors and that used the same way of passing the arguments. If you'd rather have the method take a BlockFace would be happy to change it. Then perhaps discoverNeighbors should also be changed to take a BlockFace for consistency sake.

Reply:
Oh my god I had not even noticed the other one. If you could make both accept a blockface that would be awesome